source/core/parameters.cc

```diff
@@ -186,6 +186,51 @@ namespace Parameters
     prm.leave_subsection();
   }
 
+  void
+  NonNewtonian::declare_parameters(ParameterHandler &prm)
+  {
+    prm.enter_subsection("non newtonian");
+    {
+      prm.declare_entry("model",
+                        "carreau",
+                        Patterns::Selection("carreau"),
+                        "Non newtonian model "
+                        "Choices are <carreau>.");
+      prm.declare_entry("viscosity_0",
+                        "1",
+                        Patterns::Double(),
+                        "Viscosity at rest");
+      prm.declare_entry("viscosity_inf",
+                        "2",
+                        Patterns::Double(),
+                        "Viscosity for an infinite constraint");
+      prm.declare_entry("lambda", "1", Patterns::Double(), "Relaxation time");
+      prm.declare_entry("a", "1", Patterns::Double(), "Relaxation time");
+      prm.declare_entry("n", "1", Patterns::Double(), "Power parameter");
```

Comment:
If these parameters are only for the Carreau model, I would suggest putting them in a subsection specific to the Carreau model even though at this point in time you only have this model implemented.

Reply:
Good idea @Luckabarbeau

Reply:
I fixed it in my most recent commit.